src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -4768,12 +4768,49 @@ private static OneOrMany<Symbol> CreateReducedAndFilteredSymbolsFromOriginals(Bo
 
             var methodBuilder = ArrayBuilder<Symbol>.GetInstance();
             var filteredMethodBuilder = ArrayBuilder<Symbol>.GetInstance();
+
+            // In the event of errors, IDE will often pick the first method in the group to display or navigate to.
+            // Attempt to provide a good candidate result by preferring methods that match the arguments passed into the
+            // call.  This way if there's still an overload resolution failure (say because of duplicate applicable
+            // methods), at least the first method in the group is more likely to be relevant, versus some random method
+            // in the group that wouldn't even apply.
+
+            var matchingArgumentsBuilder = ArrayBuilder<Symbol>.GetInstance();
+            var nonMatchingArgumentsBuilder = ArrayBuilder<Symbol>.GetInstance();
+
             foreach (var method in FilterOverriddenOrHiddenMethods(methods))
             {
-                AddReducedAndFilteredSymbol(methodBuilder, filteredMethodBuilder, method, typeArguments: default, receiverType, compilation);
+                var builder = argumentTypesMatchParameterTypes(method, call.Arguments)
+                    ? matchingArgumentsBuilder
+                    : nonMatchingArgumentsBuilder;
+
+                builder.Add(method);
             }
+
+            foreach (var method in matchingArgumentsBuilder)
+                AddReducedAndFilteredSymbol(methodBuilder, filteredMethodBuilder, method, typeArguments: default, receiverType, compilation);
+
+            foreach (var method in nonMatchingArgumentsBuilder)
+                AddReducedAndFilteredSymbol(methodBuilder, filteredMethodBuilder, method, typeArguments: default, receiverType, compilation);
+
+            nonMatchingArgumentsBuilder.Free();
+            matchingArgumentsBuilder.Free();
             methodBuilder.Free();
             return filteredMethodBuilder.ToOneOrManyAndFree();
+
+            static bool argumentTypesMatchParameterTypes(MethodSymbol method, ImmutableArray<BoundExpression> arguments)
```

Comment:
@dotnet/roslyn-compiler is there some existing function that would help with this.  For example, i could imagine taking advantage of refkind/optional-argument info.  

This was the simplest approach given that i don'tk now what is available in the rest of the compiler.  But if there is something i could make use of, that would be nice.